### DISCUSSION TITLE
Cannot destructure property 'RNMapboxMapsImpl' of 'undefined' as it is undefined.

### DISCUSSION THREAD

**Discussion by Charliesandewo** [rnmapbox/maps#3610] [Q&A] [answered]
Greetings everyone, i'm trying to start a project using rnmapbox/maps but while building the app i got the error in the title. 
i can't seem to find the solution to my problem at the moment, can anyone provide some guidance for me?

i just started the project using expo and the this is among the first package that i installed and i got this error.
any help is much appreciated

**Comment by uzun0ff** [comment]
Have you managed to fix this?
I am struggling with the same issue.

**Reply by Charliesandewo** [reply]
i wish i managed to fix this but sadly no. 
since then i'm working directly using Android Studio so this is on the backburner

**Reply by uzun0ff** [reply, marked as answer]
I dug a bit more and found out that eventually in order to use Mapbox, you have to create a new [development build](https://docs.expo.dev/build/setup/) through EAS and not use Expo Go.

Can't really make a stable build yet but I'm trying, hope this helps 😅

EDIT: 
@Charliesandewo I've managed to fix this.  

First you have to add the config plugin to your `app.{json,config.js,config.ts}`
```
{
  "expo": {
    "plugins": [
      [
        "@rnmapbox/maps",
        {
         "RNMapboxMapsDownloadToken": "sk.ey...qg",
          "RNMapboxMapsVersion": "11.8.0"
        }
      ]
    ]
  }
}
```

then need to either create a `development build` through EAS or locally with `expo run:ios` / `expo run:android` this will create a custom build of your application and won't use the Expo Go.

The problem is that Expo Go already has the native code built and you have to create new one with your custom native code.

Hope it's useful for you or anyone else, cheers. ✌️

**Reply by Charliesandewo** [reply]
Hi @uzun0ff just saw your solution, thanks for sharing this, just in time as i was going to continue working with this project 😅.

cheers ✌️
